istioctl/cmd/sidecar-bootstrap.go

```diff
@@ -0,0 +1,643 @@
+			istiodPort, err := deriveDiscoveryAddressPort(certs.ServiceName, certs.ServiceNamespace, kubeClient)
+			if err != nil {
+				client.Close()
+				return err
+			}
+			discoveryAddress := "istiod.istio-system.svc:" + istiodPort
+			var authPolicyStr string
+			if mutualTLS {
+				authPolicyStr = "MUTUAL_TLS"
+			} else {
+				authPolicyStr = "NONE"
+			}
+
+			meshPath := path.Join(remoteDirectory, "mesh.yaml")
+			err = remoteCopyFile([]byte(`---
+defaultConfig:
+	serviceCluster:`+certs.ServiceNamespace+`
+	controlPlaneAuthPolicy: `+authPolicyStr+`
+	discoveryAddress: `+discoveryAddress), meshPath, client)
+			if err != nil {
+				client.Close()
+				return err
+			}
+
+			//you need to deal with Sidecar CR if you want it to be "non-captured" mode
+			err = remoteRunCommand(
+				"docker run -d --name istio-proxy --network host "+
+					"-v "+meshPath+":/etc/istio/config/mesh "+
+					"-v "+remoteCaPath+":/var/run/secrets/istio/root-cert.pem "+
+					"-v "+remoteCertPath+":/var/run/secrets/istio/cert-chain.pem "+
+					"-v "+remoteKeyPath+":/var/run/secrets/istio/key.pem "+
+					"--add-host istio-pilot.istio-system.svc:"+address+" "+
+					"--add-host istiod.istio-system.svc:"+address+" "+
+					"-e POD_NAME="+certs.ServiceName+"-"+addressIdentifier+" "+
+					"-e JWT_POLICY=none "+
+					"-e PROV_CERT=/var/run/secrets/istio "+
+					"-e OUTPUT_CERTS=/var/run/secrets/istio "+
+					"-e PILOT_CERT_PROVIDER=istiod "+
+					"-e POD_NAMESPACE="+certs.ServiceNamespace+" "+
+					"-e ISTIO_META_WORKLOAD_NAME="+certs.ServiceName+" "+
+					istioProxyImage+" proxy sidecar", client)
+			if err != nil {
+				client.Close()
+				return err
+			}
+		}
+
+		client.Close()
+	}
+
+	return nil
+}
+
+func deriveSSHMethod() error {
+	if sshKeyLocation == "" {
+		term := terminal.NewTerminal(os.Stdin, "")
+		var err error
+		sshPassword, err := term.ReadPassword("Please enter the SSH password: ")
+		if err != nil {
+			return err
+		}
+		if sshPassword == "" {
+			return fmt.Errorf("a password, or SSH key location is required for sidecar-bootstrap")
+		}
+		sshAuthMethod = ssh.Password(sshPassword)
+	} else {
+		// Attempt to parse the key.
+		rawKey, err := ioutil.ReadFile(sshKeyLocation)
+		if err != nil {
+			return err
+		}
+		key, err := ssh.ParsePrivateKey(rawKey)
+		if err != nil {
+			if err, ok := err.(*ssh.PassphraseMissingError); ok {
+				term := terminal.NewTerminal(os.Stdin, "")
+				sshKeyPassword, err := term.ReadPassword("Please enter the password for the SSH key: ")
+				if err != nil {
+					return err
+				}
+				decryptedKey, err := ssh.ParsePrivateKeyWithPassphrase(rawKey, []byte(sshKeyPassword))
+				if err != nil {
+					return err
+				}
+				sshAuthMethod = ssh.PublicKeys(decryptedKey)
+			} else {
+				return err
+			}
+		} else {
+			sshAuthMethod = ssh.PublicKeys(key)
+		}
+	}
+
+	return nil
+}
+
+func vmBootstrapCommand() *cobra.Command {
+	vmBSCommand := &cobra.Command{
+		Use:   "sidecar-bootstrap",
```

Comment:
In Cobra we don't just give the command, if it has a positional argument we show it.  So "sidecar-bootstrap <workload>" (or something like that.)